client/src/controllers/ClipboardController.ts

```diff
@@ -0,0 +1,35 @@
+import { Controller } from '@hotwired/stimulus';
+
+export class ClipboardController extends Controller<HTMLElement> {
+  static targets = ['value'];
+
+  declare valueTarget:
+    | HTMLInputElement
+    | HTMLTextAreaElement
+    | HTMLSelectElement;
+
+  copy() {
+    this.dispatch('start');
+    if (!navigator.clipboard) {
+      this.showError();
+      return;
+    }
+
+    const value = this.valueTarget.value;
```

Comment:
I think it may make more sense to check for no value right at the start of the method, returning early before we even check for the clipboard global, before we dispatch `copy`.

Reply:
yes